### PR TITLE
Extend remove repository to remove plugin repositories

### LINKED ISSUE #2709: Create RemovePluginRepository recipe
This should be very similar to the [RemoveRepository](https://github.com/openrewrite/rewrite/blob/main/rewrite-maven/src/main/java/org/openrewrite/maven/RemoveRepository.java) recipe

Sample: 

```xml

    <pluginRepositories>
        <pluginRepository>
            <id>central</id>
            <url>https://repo1.maven.org/maven2//</url>
        </pluginRepository>

          <pluginRepository>
            <id>atlassian</id>
            <url>https://packages.atlassian.com/mvn/maven-external/</url>
        </pluginRepository>
    </pluginRepositories>

```

### PR DESCRIPTION
**Pull request by yeikel**
Closes #2709

### CODE REVIEW

File: rewrite-maven/src/test/java/org/openrewrite/maven/RemoveRepositoryTest.java [hunk 1]
```diff
@@ -93,4 +93,77 @@ void removeSimpleRepoWithUrlOnly() {
...
+                <pluginRepositories>
+                  <pluginRepository>
+                    <id>myRepo</id>
+                    <url>http://myrepo.maven.com/repo</url>
+                  </pluginRepository>
+                  <pluginRepository>
+                    <id>identicalRepo</id>
+                    <url>http://myrepo.maven.com/repo</url>
+                  </pluginRepository>
+                </pluginRepositories>
```
Comment (yeikel) [inline rewrite-maven/src/test/java/org/openrewrite/maven/RemoveRepositoryTest.java:120]: I maintained the same behavior, but I don't agree with the existing functionality

In this case, the user provided the repo id `myRepo` but this recipe removes also the other repo with id `identicalRepo`

Even if it is the same URL, I don't think that this recipe should do that

File: rewrite-maven/src/test/java/org/openrewrite/maven/RemoveRepositoryTest.java [hunk 2]
```diff
@@ -98,6 +98,54 @@ void removeSimpleRepoWithUrlOnly() {
...
+              """,
+            """
+             <project>
+               <groupId>com.mycompany.app</groupId>
+               <artifactId>my-app</artifactId>
+               <version>1</version>
+               <pluginRepositories>
+               </pluginRepositories>
+                <repositories>
+               </repositories>
```
Comment (yeikel) [inline rewrite-maven/src/test/java/org/openrewrite/maven/RemoveRepositoryTest.java:137]: Should we have the option to allow users to pick? Or is it ok to remove from both as a default?

Comment (timtebeek) [reply rewrite-maven/src/test/java/org/openrewrite/maven/RemoveRepositoryTest.java:137]: I think it's fine to remove both, or at least until a compelling argument to introduce a parameter is made in a new issue. The use cases I see for removing repositories is typically when repositories are decommissioned; in that case it makes sense to remove both the regular repository and the plugin repository at the same time.

File: rewrite-maven/src/test/java/org/openrewrite/maven/RemoveRepositoryTest.java (hunk 1 above)
Comment (sambsnyd) [reply rewrite-maven/src/test/java/org/openrewrite/maven/RemoveRepositoryTest.java:120]: @yeikel I agree. It feels like a bug that the recipe would remove a repository with an ID other than the one the user specified.

Comment (yeikel) [reply rewrite-maven/src/test/java/org/openrewrite/maven/RemoveRepositoryTest.java:120]: I'll fix it before it is merged. Thank you for your review

Comment (yeikel) [reply rewrite-maven/src/test/java/org/openrewrite/maven/RemoveRepositoryTest.java]: @sambsnyd I updated it with 8dc10b861b38c20b6a0711fb49daff416cfd0691 to only remove if both match

One additional detail that I noticed about this recipe is that it contains additional keys that are not necessarily used in the recipe

image


I tentatively pushed 379aa19194cdde9f6398d2a85d6876b81c811dd0 to remove them and simplify them, but we can revert it if you think that it is needed.

Comment (jkschneider) [reply rewrite-maven/src/test/java/org/openrewrite/maven/RemoveRepositoryTest.java]: Thanks for cleaning up the options as well!

**Pull request merged**